### PR TITLE
Bug 1580778 - Fix version names

### LINKED ISSUE #5558: [Bug] Nightly build from Play Store has quotation marks in its name
## Steps to reproduce
1. Install Firefox Preview Nightly build from Play Store.
2. Go to Settings - About Firefox Nightly and observe the name of the build.

### Expected behavior
The name should be without quotation marks.

### Actual behavior
The name of the build is between quotation marks.

### Device information
* Android device: all
* Fenix version: Nightly 190924 from Play Store

picturemessage_4tp3wkh5 gqb

### LINKED ISSUE #5615: Releases don't have a versionName
Spotted in automation testing, release builds don't have `-PversionName` provided to them - here's what my `gradle` command looks like when testing locally:
```
./gradlew clean assembleGeckoBetaFenixProduction
```

@JohanLorenzo are you able to take a look at this?

### PR DESCRIPTION
**Pull request by JohanLorenzo**
Fixes #5558 
Fixes #5615

This patch just changes the nightly, the beta, and the production graphs this way:

## Nightly

Note the shell quotes are now fine. I double checked by running the command manually and making sure the content of the APK has the right version name.
```py
[
    "change", 
    [
        "build-nightly-legacy", 
        "task", 
        "payload", 
        "command", 
        8
    ], 
    [
        "automation/taskcluster/helper/get-secret.py -s project/mobile/fenix/nightly-legacy -k adjust -f .adjust_token && automation/taskcluster/helper/get-secret.py -s project/mobile/fenix/nightly-legacy -k firebase -f app/src/nightly-legacy/res/values/firebase.xml && automation/taskcluster/helper/get-secret.py -s project/mobile/fenix/nightly-legacy -k digital_asset_links -f .digital_asset_links_token && automation/taskcluster/helper/get-secret.py -s project/mobile/fenix/nightly-legacy -k leanplum -f .leanplum_token && automation/taskcluster/helper/get-secret.py -s project/mobile/fenix/nightly-legacy -k sentry_dsn -f .sentry_token && ./gradlew clean assembleGeckoNightlyFenixNightlyLegacy '-PversionName=\"Nightly 190922 18:01\"'", 
        "automation/taskcluster/helper/get-secret.py -s project/mobile/fenix/nightly-legacy -k adjust -f .adjust_token && automation/taskcluster/helper/get-secret.py -s project/mobile/fenix/nightly-legacy -k firebase -f app/src/nightly-legacy/res/values/firebase.xml && automation/taskcluster/helper/get-secret.py -s project/mobile/fenix/nightly-legacy -k digital_asset_links -f .digital_asset_links_token && automation/taskcluster/helper/get-secret.py -s project/mobile/fenix/nightly-legacy -k leanplum -f .leanplum_token && automation/taskcluster/helper/get-secret.py -s project/mobile/fenix/nightly-legacy -k sentry_dsn -f .sentry_token && ./gradlew clean assembleGeckoNightlyFenixNightlyLegacy '-PversionName=Nightly 190922 18:01'"
    ]
]
[
    "change", 
    [
        "build-nightly", 
        "task", 
        "payload", 
        "command", 
        8
    ], 
    [
        "automation/taskcluster/helper/get-secret.py -s project/mobile/fenix/nightly -k adjust -f .adjust_token && automation/taskcluster/helper/get-secret.py -s project/mobile/fenix/nightly -k firebase -f app/src/nightly/res/values/firebase.xml && automation/taskcluster/helper/get-secret.py -s project/mobile/fenix/nightly -k digital_asset_links -f .digital_asset_links_token && automation/taskcluster/helper/get-secret.py -s project/mobile/fenix/nightly -k leanplum -f .leanplum_token && automation/taskcluster/helper/get-secret.py -s project/mobile/fenix/nightly -k sentry_dsn -f .sentry_token && ./gradlew clean assembleGeckoNightlyFenixNightly '-PversionName=\"Nightly 190922 18:01\"'", 
        "automation/taskcluster/helper/get-secret.py -s project/mobile/fenix/nightly -k adjust -f .adjust_token && automation/taskcluster/helper/get-secret.py -s project/mobile/fenix/nightly -k firebase -f app/src/nightly/res/values/firebase.xml && automation/taskcluster/helper/get-secret.py -s project/mobile/fenix/nightly -k digital_asset_links -f .digital_asset_links_token && automation/taskcluster/helper/get-secret.py -s project/mobile/fenix/nightly -k leanplum -f .leanplum_token && automation/taskcluster/helper/get-secret.py -s project/mobile/fenix/nightly -k sentry_dsn -f .sentry_token && ./gradlew clean assembleGeckoNightlyFenixNightly '-PversionName=Nightly 190922 18:01'"
    ]
]
[
    "change", 
    [
        "build-fennec-production", 
        "task", 
        "payload", 
        "command", 
        8
    ], 
    [
        "automation/taskcluster/helper/get-secret.py -s project/mobile/fenix/fennec-production -k adjust -f .adjust_token && automation/taskcluster/helper/get-secret.py -s project/mobile/fenix/fennec-production -k firebase -f app/src/fennec-production/res/values/firebase.xml && automation/taskcluster/helper/get-secret.py -s project/mobile/fenix/fennec-production -k digital_asset_links -f .digital_asset_links_token && automation/taskcluster/helper/get-secret.py -s project/mobile/fenix/fennec-production -k leanplum -f .leanplum_token && automation/taskcluster/helper/get-secret.py -s project/mobile/fenix/fennec-production -k sentry_dsn -f .sentry_token && ./gradlew clean assembleGeckoBetaFennecProduction", 
        "automation/taskcluster/helper/get-secret.py -s project/mobile/fenix/fennec-production -k adjust -f .adjust_token && automation/taskcluster/helper/get-secret.py -s project/mobile/fenix/fennec-production -k firebase -f app/src/fennec-production/res/values/firebase.xml && automation/taskcluster/helper/get-secret.py -s project/mobile/fenix/fennec-production -k digital_asset_links -f .digital_asset_links_token && automation/taskcluster/helper/get-secret.py -s project/mobile/fenix/fennec-production -k leanplum -f .leanplum_token && automation/taskcluster/helper/get-secret.py -s project/mobile/fenix/fennec-production -k sentry_dsn -f .sentry_token && ./gradlew clean assembleGeckoBetaFennecProduction '-PversionName=Nightly 190922 18:01'"
    ]
]
```

## Beta

```py
[
    "change", 
    [
        "build-production", 
        "task", 
        "payload", 
        "command", 
        8
    ], 
    [
        "automation/taskcluster/helper/get-secret.py -s project/mobile/fenix/production -k adjust -f .adjust_token && automation/taskcluster/helper/get-secret.py -s project/mobile/fenix/production -k firebase -f app/src/production/res/values/firebase.xml && automation/taskcluster/helper/get-secret.py -s project/mobile/fenix/production -k digital_asset_links -f .digital_asset_links_token && automation/taskcluster/helper/get-secret.py -s project/mobile/fenix/production -k leanplum -f .leanplum_token && automation/taskcluster/helper/get-secret.py -s project/mobile/fenix/production -k sentry_dsn -f .sentry_token && ./gradlew clean assembleGeckoBetaFenixProduction", 
        "automation/taskcluster/helper/get-secret.py -s project/mobile/fenix/production -k adjust -f .adjust_token && automation/taskcluster/helper/get-secret.py -s project/mobile/fenix/production -k firebase -f app/src/production/res/values/firebase.xml && automation/taskcluster/helper/get-secret.py -s project/mobile/fenix/production -k digital_asset_links -f .digital_asset_links_token && automation/taskcluster/helper/get-secret.py -s project/mobile/fenix/production -k leanplum -f .leanplum_token && automation/taskcluster/helper/get-secret.py -s project/mobile/fenix/production -k sentry_dsn -f .sentry_token && ./gradlew clean assembleGeckoBetaFenixProduction -PversionName=1.2.3-beta.4"
    ]
]
```

## Production

```py
[
    "change", 
    [
        "build-production", 
        "task", 
        "payload", 
        "command", 
        8
    ], 
    [
        "automation/taskcluster/helper/get-secret.py -s project/mobile/fenix/production -k adjust -f .adjust_token && automation/taskcluster/helper/get-secret.py -s project/mobile/fenix/production -k firebase -f app/src/production/res/values/firebase.xml && automation/taskcluster/helper/get-secret.py -s project/mobile/fenix/production -k digital_asset_links -f .digital_asset_links_token && automation/taskcluster/helper/get-secret.py -s project/mobile/fenix/production -k leanplum -f .leanplum_token && automation/taskcluster/helper/get-secret.py -s project/mobile/fenix/production -k sentry_dsn -f .sentry_token && ./gradlew clean assembleGeckoBetaFenixProduction", 
        "automation/taskcluster/helper/get-secret.py -s project/mobile/fenix/production -k adjust -f .adjust_token && automation/taskcluster/helper/get-secret.py -s project/mobile/fenix/production -k firebase -f app/src/production/res/values/firebase.xml && automation/taskcluster/helper/get-secret.py -s project/mobile/fenix/production -k digital_asset_links -f .digital_asset_links_token && automation/taskcluster/helper/get-secret.py -s project/mobile/fenix/production -k leanplum -f .leanplum_token && automation/taskcluster/helper/get-secret.py -s project/mobile/fenix/production -k sentry_dsn -f .sentry_token && ./gradlew clean assembleGeckoBetaFenixProduction -PversionName=1.2.3"
    ]
]

```

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- [ ] **Milestone**: Make sure issues finished by this pull request are added to the [milestone](https://github.com/mozilla-mobile/fenix/milestones) of the version currently in development.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: taskcluster/fenix_taskgraph/transforms/build.py [hunk 1]
```diff
@@ -83,7 +84,7 @@ def add_nightly_version(config, tasks):
 
     for task in tasks:
         if task.pop("include-nightly-version", False):
-            task["run"]["gradlew"].append('-PversionName="{}"'.format(formated_date_time))
+            task["run"]["gradlew"].append('-PversionName={}'.format(formated_date_time))
```
Comment (JohanLorenzo) [inline taskcluster/fenix_taskgraph/transforms/build.py]: It turns out shell_quote handles spaces pretty well. So no need for extra double quotes, here.

https://github.com/mozilla-mobile/fenix/blob/3f41a4e9c63ffce45e4038b5f266613bccd88b56/taskcluster/fenix_taskgraph/job.py#L68

File: taskcluster/fenix_taskgraph/transforms/build.py [hunk 2]
```diff
@@ -9,6 +9,7 @@
 from __future__ import absolute_import, print_function, unicode_literals
 
 import datetime
+import os
```
Comment (JohanLorenzo) [inline taskcluster/fenix_taskgraph/transforms/build.py]: `add_release_version()` calls `os.environ`. It shows `add_release_version()` wasn't called before.

File: taskcluster/fenix_taskgraph/transforms/build.py [hunk 3]
```diff
@@ -92,10 +93,10 @@ def add_release_version(config, tasks):
     for task in tasks:
         if task.pop("include-release-version", False):
             # TODO Move GIT_TAG to a parameter
-            git_tag = os.environ['GIT_TAG']
+            git_tag = os.environ.get('GIT_TAG', 'v0')
```
Comment (mitchhentges) [inline taskcluster/fenix_taskgraph/transforms/build.py]: Hmm, I wonder if `v0` will be confused with a real tag, perhaps we should make the default into something that is more clearly a "not-set" default?

Comment (JohanLorenzo) [reply taskcluster/fenix_taskgraph/transforms/build.py]: Good call. Done.

**Comment by sblatz** [comment]
Rebasing now that master builds 😬

**Pull request merged**